pkg/client/aad.go

```diff
@@ -15,6 +15,9 @@ import (
 	"github.com/sirupsen/logrus"
 )
 
+// can instantiate this with a mock in unit tests
+var newTokenAcquirer = newAadTokenAcquirer
```

Comment:
spoke with @bcho - we actually want to try to avoid this pattern of using/overriding global variables for this sort of thing - could we do something like this instead?


`type aadTokenAcquirerFunc func(authority, clientID, clientSecret string) (confidential.AuthResult, error)

type aadClientImpl struct {
   ....
   tokenAcquirerFunc aadTokenAcquirerFunc
}

func newAADClient(..., acquirerFunc aadTokenAcquirerFunc) ... {
      impl.aadTokenAcquirerFunc = acquirerFunc
}

`